app/models/user.rb

```diff
@@ -11,10 +11,17 @@ class User < ApplicationRecord
 
   enum :role, { member: "member", admin: "admin" }, validate: true
 
+  has_one_attached :profile_image
+  validate :profile_image_size
+
   generates_token_for :password_reset, expires_in: 15.minutes do
     password_salt&.last(10)
   end
 
+  def profile_image_thumbnail
+    profile_image.variant(resize_to_limit: [150, 150])
```

Comment:
2 notes here:

1. Can we get this added to the sidebar and within settings menu?

![CleanShot 2024-04-29 at 09 42 45](https://github.com/maybe-finance/maybe/assets/16676157/136676a2-0ad2-4535-beb9-9f8e2bcacff7)


3. We could probably add `preprocessed: true` here since this shows in sidebar and we're confident it will always need to load

Reply:
Done this but not sure what where I would add `preprocessed: true`.

How about adding `.processed` to the model method?

```ruby
  def profile_image_thumbnail
    profile_image.variant(resize_to_limit: [150, 150]).processed
  end
  ```

Reply:
I was just referring to the Rails guide example here:

https://guides.rubyonrails.org/active_storage_overview.html#:~:text=If%20you%20know%20in%20advance%20that%20your%20variants%20will%20be%20accessed%2C%20you%20can%20specify%20that%20Rails%20should%20generate%20them%20ahead%20of%20time%3A

Reply:
This is a better way to do it. Nice one 👍🏻